tests/db/test_unit_db_models.py

```diff
@@ -0,0 +1,667 @@
+    assert len(db_input_dataset.list_jobs_input) == N_JOBS
+    assert len(db_input_dataset.list_jobs_output) == 0
+    assert len(db_output_dataset.list_jobs_input) == 0
+    assert len(db_output_dataset.list_jobs_output) == N_JOBS
+
+
+async def test_project_name_not_unique(MockCurrentUser, db, project_factory):
+    """
+    GIVEN the fractal_server database
+    WHEN I create two projects with the same name and same user
+    THEN no exception is raised
+    """
+    PROJ_NAME = "project name"
+    async with MockCurrentUser(persist=True) as user:
+        p0 = await project_factory(user, name=PROJ_NAME)
+        p1 = await project_factory(user, name=PROJ_NAME)
+
+    stm = select(Project).where(Project.name == PROJ_NAME)
+    res = await db.execute(stm)
+    project_list = res.scalars().all()
+    assert len(project_list) == 2
+    assert p0 in project_list
+    assert p1 in project_list
+
+
+async def test_task_workflow_association(
+    db, project_factory, MockCurrentUser, task_factory
+):
+    async with MockCurrentUser(persist=True) as user:
+        project = await project_factory(user)
+        t0 = await task_factory(source="source0")
+        t1 = await task_factory(source="source1")
+
+        wf = Workflow(name="my wfl", project_id=project.id)
+        args = dict(arg="test arg")
+        await wf.insert_task(t0.id, db=db, args=args)
+
+        db.add(wf)
+        await db.commit()
+        await db.refresh(wf)
+
+        debug(wf)
+        assert wf.task_list[0].args == args
+        # check workflow
+        assert len(wf.task_list) == 1
+        assert wf.task_list[0].id == t0.id
+        # check association table
+        stm = (
+            select(WorkflowTask)
+            .where(WorkflowTask.workflow_id == wf.id)
+            .where(WorkflowTask.task_id == t0.id)
+        )
+        res = await db.execute(stm)
+        link = res.scalars().one()
+        debug(link)
+        assert link.task_id == t0.id
+
+        # Insert at position 0
+        await wf.insert_task(t1.id, order=0, db=db)
+        db.add(wf)
+        await db.commit()
+        await db.refresh(wf)
+
+        stm = (
+            select(WorkflowTask)
+            .where(WorkflowTask.workflow_id == wf.id)
+            .where(WorkflowTask.task_id == t1.id)
+        )
+        res = await db.execute(stm)
+        link = res.scalars().one()
+        debug(link)
+        assert link.order == 0
+        assert link.task_id == t1.id
+
+
+async def test_workflow_insert_task_with_args_schema(
+    db, project_factory, MockCurrentUser, task_factory
+):
+    """
+    GIVEN a Workflow, and a Task with valid args_schema (including defaults)
+    WHEN the Task is inserted into the Workflow
+    THEN the WorkflowTask.args attribute is set correctly
+    """
+    from pydantic import BaseModel
+    from typing import Optional
+
+    async with MockCurrentUser(persist=True) as user:
+
+        # Prepare models to generate a valid JSON Schema
```

Comment:
I suggest we use a hard-coded JSON schema, to avoid issues when we'll switch to Pydantic v2.

Here is the one that is currently produced here:
```json
{'title': '_Arguments', 'type': 'object', 'properties': {'arg_no_default': {'title': 'Arg No Default', 'type': 'integer'
        }, 'arg_default_one': {'title': 'Arg Default One', 'default': 'one', 'type': 'string'
        }, 'arg_default_none': {'title': 'Arg Default None', 'type': 'string'
        }, 'innerA': {'$ref': '#/definitions/_InnerArgument'
        }, 'innerB': {'title': 'Innerb', 'default': {'x': 11, 'y': 2
            }, 'allOf': [
                {'$ref': '#/definitions/_InnerArgument'
                }
            ]
        }
    }, 'required': ['arg_no_default', 'innerA'
    ], 'definitions': {'_InnerArgument': {'title': '_InnerArgument', 'type': 'object', 'properties': {'x': {'title': 'X', 'type': 'integer'
                }, 'y': {'title': 'Y', 'default': 2, 'type': 'integer'
                }
            }, 'required': ['x'
            ]
        }
    }
}
```